### PR TITLE
fix settings explorer

### PR DESCRIPTION
**Pull request by SamSaffron**
- FIX: setting explorer was exceeding token budget
- In dev it is important to tell when calls are made to open ai this adds a console log to increase awareness around token usage
- PERF: stop counting tokens so often
- bug fix, commands may reach in for tokenizer
- add logging to console for anthropic calls as well

### CODE REVIEW

File: lib/shared/inference/anthropic_completions.rb [hunk 1]
```diff
@@ -104,17 +105,22 @@ def self.perform!(
...
           end
+        ensure
+          if block_given?
+            log.update!(
+              raw_response_payload: response_raw,
+              request_tokens: DiscourseAi::Tokenizer::AnthropicTokenizer.size(prompt),
+              response_tokens: DiscourseAi::Tokenizer::AnthropicTokenizer.size(response_data),
+            )
+          end
+          if Rails.env == "development" && log
```
Comment (martin-brennan) [inline lib/shared/inference/anthropic_completions.rb:121]:
```suggestion
          if Rails.env.development? && log
```

File: lib/shared/inference/openai_completions.rb [hunk 2]
```diff
@@ -125,19 +126,25 @@ def self.perform!(
...
+        if log && block_given?
+          request_tokens = DiscourseAi::Tokenizer::OpenAiTokenizer.size(extract_prompt(messages))
+          response_tokens = DiscourseAi::Tokenizer::OpenAiTokenizer.size(response_data)
+          log.update!(
+            raw_response_payload: response_raw,
+            request_tokens: request_tokens,
+            response_tokens: response_tokens,
+          )
+        end
+        if log && Rails.env == "development"
```
Comment (martin-brennan) [inline lib/shared/inference/openai_completions.rb:145]:
```suggestion
        if log && Rails.env.development?
```

File: lib/modules/ai_bot/commands/search_settings_command.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    end
+
+    INCLUDE_DESCRIPTIONS_MAX_LENGTH = 10
+    MAX_RESULTS = 200
+
+    def process(query:)
+      @last_query = query
+      @last_num_results = 0
+
+      terms = query.split(",").map(&:strip).map(&:downcase).reject(&:blank?)
```
Comment (martin-brennan) [inline lib/modules/ai_bot/commands/search_settings_command.rb:42]: Maybe we can add  min length and early return? Pointless to allow people to use up CPU time searching for `a` :)

File: lib/shared/inference/anthropic_completions.rb (hunk 1 above)
Comment (SamSaffron) [reply lib/shared/inference/anthropic_completions.rb:121]: yeah people like this incantation, but I really dislike it for some reason

File: lib/modules/ai_bot/commands/search_settings_command.rb (hunk 3 above)
Comment (SamSaffron) [reply lib/modules/ai_bot/commands/search_settings_command.rb:42]: I think a is OK, its not chewing up too much cpu... this is going to have to scan through everything no matter what you give it... no indexes.

File: lib/shared/inference/anthropic_completions.rb (hunk 1 above)
Comment (SamSaffron) [reply lib/shared/inference/anthropic_completions.rb:121]: ok I dug into it... and yeah it is unconditionally better cause it is faster :open_mouth: 

https://github.com//blob/bb52c41dbfdf8d57b58f82319506446b70bca1b1/activesupport/lib/active_support/environment_inquirer.rb#L27-L33

**Review by martin-brennan — APPROVED** [review]
Works really nice locally :ok_hand:

File: lib/modules/ai_bot/commands/search_settings_command.rb (hunk 3 above)
Comment (martin-brennan) [reply lib/modules/ai_bot/commands/search_settings_command.rb:42]: Yeah I just mean it's kind of pointless to even try with anything < 3 characters long...but if you think it's fine just leave it :+1:

**Pull request merged**